src/msw/desktopenv.cpp

```diff
@@ -0,0 +1,54 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        src/msw/desktopenv.cpp
+// Purpose:     implementation of wxDesktopEnvBase
+// Author:      Igor Korot
+// Created:     2021-09-05
+// Copyright:   (c) 2021 wxWidgets development team
+// Licence:     wxWidgets licence
+///////////////////////////////////////////////////////////////////////////////
+
+// ============================================================================
+// declarations
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// headers
+// ----------------------------------------------------------------------------
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#ifdef __BORLANDC__
+#pragma hdrstop
+#endif
+
+#include "wx/log.h"
+#include "wx/desktopenv.h"
+#include "wx/msw/private.h"
+
+// ============================================================================
+// implementation
+// ============================================================================
+
+/* static */
+bool wxDesktopEnv::MoveToRecycleBin(const wxString &path)
+{
+    bool res = wxDesktopEnvBase::MoveToRecycleBin( path );
+    if( res )
+    {
+        SHFILEOPSTRUCT fileOp;
+        wxZeroMemory( fileOp );
+        fileOp.hwnd = NULL;
+        fileOp.wFunc = FO_DELETE;
+        wxString temp = path + '\0';
+        fileOp.pFrom = temp.t_str();
+        fileOp.pTo = NULL;
+        fileOp.fFlags = FOF_ALLOWUNDO | FOF_NOERRORUI | FOF_NOCONFIRMATION | FOF_SILENT;
+        res = SHFileOperation( &fileOp );
+        if( res != 0 )
+            wxLogSysError( _( "Failed to move '%s' to Recycle Bin" ), path );
```

Comment:
`wxLogSysError()` should not be used here: `SHFileOperation()` does not affect the last error code.

Either just display the simple error message or if you want to display the specific error, you could do something like this
```cpp
wxLogError(_( "Moving '%s' to Recycle Bin failed with error 0x%08lx (%s)."),
    path, (long)res, wxSysErrorMsgStr(res));
```